src/libraries/Descriptor.sol

```diff
@@ -0,0 +1,520 @@
+        if (quotesCount > 0) {
+            // create a new bytes array with enough space to hold the original bytes plus space for the backslashes to escape the quotes
+            bytes memory escapedBytes = new bytes(symbolBytes.length + quotesCount);
+            uint256 index;
+            for (uint8 i = 0; i < symbolBytes.length; i++) {
+                // add a '\' before any double quotes
+                if (symbolBytes[i] == '"') {
+                    escapedBytes[index++] = "\\";
+                }
+                // copy each byte from original string to the new array
+                escapedBytes[index++] = symbolBytes[i];
+            }
+            return string(escapedBytes);
+        }
+        return symbol;
+    }
+
+    /// @notice Generates the first part of the description for a Uniswap V4 NFT
+    /// @param quoteCurrencySymbol The symbol of the quote currency
+    /// @param baseCurrencySymbol The symbol of the base currency
+    /// @param poolManager The address of the pool manager
+    /// @param quoteCurrency The address of the quote currency
+    /// @return The first part of the description
+    function generateDescriptionPartOne(
+        string memory quoteCurrencySymbol,
+        string memory baseCurrencySymbol,
+        string memory poolManager,
+        string memory quoteCurrency
+    ) private pure returns (string memory) {
+        // displays quote currency first, then base currency
+        return string(
+            abi.encodePacked(
+                "This NFT represents a liquidity position in a Uniswap V4 ",
+                quoteCurrencySymbol,
+                "-",
+                baseCurrencySymbol,
+                " pool. ",
+                "The owner of this NFT can modify or redeem the position.\\n",
+                "\\nPool Manager Address: ",
+                poolManager,
+                "\\n",
+                quoteCurrencySymbol,
+                " Address: ",
+                quoteCurrency,
+                "\\n"
+            )
+        );
+    }
+
+    /// @notice Generates the second part of the description for a Uniswap V4 NFTs
+    /// @param tokenId The token ID
+    /// @param baseCurrencySymbol The symbol of the base currency
+    /// @param baseCurrency The address of the base currency
+    /// @param hooks The address of the hooks contract
+    /// @param feeTier The fee tier of the pool
+    /// @return The second part of the description
+    function generateDescriptionPartTwo(
+        string memory tokenId,
+        string memory baseCurrencySymbol,
+        string memory baseCurrency,
+        string memory hooks,
+        string memory feeTier
+    ) private pure returns (string memory) {
+        return string(
+            abi.encodePacked(
+                baseCurrencySymbol,
+                " Address: ",
+                baseCurrency,
+                "\\nHook Address: ",
+                hooks,
+                "\\nFee Tier: ",
+                feeTier,
```

Comment:
what does it display for dynamic fees?

Reply:
good point. should i ask phil? maybe if its equal to DYNAMIC_FEE_FLAG, display DYNAMIC? or else it could aways be changing